### DISCUSSION TITLE
problem installing new binary

### DISCUSSION THREAD

**Discussion by steforni** [cardonabits/haxo-hw#91] [Q&A] [answered]
hello everyone
first I join the long list of people excited about this fantastic project and the beautiful community behind it!!!
I ask for your help because I am having problems with the installation of the new binary.
The problems arise when I run "sudo cp haxo001_new /usr/local/bin/haxo001" and I get the following error: file text busy which would seem to be related to the fact that the file to be replaced is running
Looking online I found some possible solutions such as using cp -f to force the operation and in fact then it goes without giving me any error
Unfortunately after finishing all the procedure however the haxophone doesn't work anymore, so this forcing caused something to break.
Could the error “file text busy” be related to the fact that I work from a windows operating system? (I know I am a bad person, forgive me).
Now I'm stuck because it doesn't work and I don't even know how to get back except by formatting the sd card and reinstalling everything (already done once)
if anyone can help me I would be eternally grateful!! :)
Ciao 
Stefano

**Comment by jcard0na** [comment]
Hi @steforni,

Are you following the instructions from [this section of the manual](https://github.com/cardonabits/haxo-hw/blob/main/docs/manual/manual.md#install-the-newest-binary)?  If so, I suspect that they might be incomplete:  you seem to be trying to overwrite a file that is running.  Before you can overwrite `haxo001` you would need to stop the haxophone service.
The simplest way to do that is:

```
sudo systemctl stop haxo
```

after that, you should be able to copy the file.   Please let me know if that works, and I will update the documentation accordingly!

Cheers,

PS.  No need to apologize for using Windows... 😆

**Reply by steforni** [reply]
hello javier
thank you for the super fast answer!
I followed your suggestion and the installation procedure of the new binary continued without errors, so apparently it worked...but...when I try to play the haxophone it sounds strange (with very low notes) and the instrument change and transpose functions do not work :(
i attach you the logs of haxo service

pi@haxophone:~ $ journalctl -u haxo
Boot 23f5ccd22f674322a70604d7f4547235 --
Dec 21 20:29:15 haxophone systemd[1]: Started haxophone.
Dec 21 20:29:21 haxophone haxo001[504]: fluidsynth: warning: SDL2 not initialized, SDL2 audio driver won't be usable
Dec 21 20:29:21 haxophone haxo001[504]: [2023-12-21T20:29:21Z INFO  haxo001::alsa] Found alsa card MAX98357A
Dec 21 20:29:32 haxophone haxo001[504]: Synth createdDec 21 18:26:32 haxophone haxo001[508]: fluidsynth: warning: SDL2 not initialized, SDL2 audio driver won't be usable
Dec 21 18:26:32 haxophone haxo001[508]: [2023-12-21T18:26:32Z INFO  haxo001::alsa] Found alsa card MAX98357A
Dec 21 18:26:42 haxophone haxo001[508]: Synth created
Dec 21 18:26:43 haxophone haxo001[508]: [2023-12-21T18:26:43Z INFO  haxo001::midi]
Dec 21 18:26:43 haxophone haxo001[508]:     Available output ports:
Dec 21 18:26:43 haxophone haxo001[508]: [2023-12-21T18:26:43Z INFO  haxo001::midi] 0: Midi Through:Midi Through Port-0>
Dec 21 18:26:43 haxophone haxo001[508]: [2023-12-21T18:26:43Z INFO  haxo001::midi] 1: f_midi:f_midi 24:0
Dec 21 18:26:43 haxophone haxo001[508]: [2023-12-21T18:26:43Z INFO  haxo001::midi] Picked the last one
Dec 21 18:26:43 haxophone haxo001[508]: Starting haxophone...
Dec 21 20:28:17 haxophone systemd[1]: Stopping haxophone...
Dec 21 20:28:18 haxophone systemd[1]: haxo.service: Succeeded.
Dec 21 20:28:18 haxophone systemd[1]: Stopped haxophone.
Dec 21 20:28:18 haxophone systemd[1]: haxo.service: Consumed 20min 30.725s CPU time.

**Reply by jcard0na** [reply, marked as answer]
Hi @steforni,

Good to know you were able to copy the binary.

Regarding transposition, the feature introduced a change in `notemap.json`, the file that maps fingerings to midi notes.  See https://github.com/cardonabits/haxo-rs/pull/27 for details.

The fix would be to fetch the [latest notemap.json from the repo](https://raw.githubusercontent.com/cardonabits/haxo-rs/refs/heads/main/notemap.json) and copy it over the one in the released image.  You can use the same method you did for `haxo001`.

That should fix the "very low notes" problem.

Regarding the transposition fingerings, please see https://github.com/cardonabits/haxo-rs/pull/28  You need to draw air on a `low B` fingering, which is different from the `low Bb` fingering you use to switch instruments.  Can you check?

Cheers,

**Reply by steforni** [reply]
Hi
I changed notemap.json and now everything works perfectly!!! I'm super happy and excited, thank you very much :)
as a small contribution to the project, I'd like to share the sequence of operations I did (after some failed attempts) so that if someone inexperienced like me should have the same problem, he can try to follow them: 
1. Connect via ssh and disable overlay, reboot 
2. Stop the haxo service:
       `sudo systemctl stop haxo`
3. Download the notemap.json file on your local pc (I found here https://github.com/cardonabits/haxo-rs)  or laptop and use scp to copy it to the haxophone:
       `scp notemap.json pi@haxophone.local:/home/pi`
4. replace the file with the new version
      `sudo cp notemap.json /usr/share/haxo/notemap.json`
5. reboot sudo reboot
6. Re-enable overlay
In my case this sequence worked well, if I wrote something wrong or improvable clearly correct me :)
greetings to all

**Reply by jcard0na** [reply]
Hi @steforni 

Thank you for reporting back and sharing your detailed solution.  It is unlikely that `notemap.json` will change again, so in future releases you'll probably just need to update the binary (`haxo001`).

Best,
